deployment/src/main/java/org/wso2/carbon/testgrid/deployment/deployers/PuppetDeployer.java

```diff
@@ -0,0 +1,65 @@
+package org.wso2.carbon.testgrid.deployment.deployers;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.testgrid.common.DeployerService;
+import org.wso2.carbon.testgrid.common.Deployment;
+import org.wso2.carbon.testgrid.common.Utils;
+import org.wso2.carbon.testgrid.common.exception.TestGridDeployerException;
+import org.wso2.carbon.testgrid.deployment.DeployerConstants;
+import org.wso2.carbon.testgrid.deployment.DeploymentUtil;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class PuppetDeployer implements DeployerService {
+    private static final Log log = LogFactory.getLog(PuppetDeployer.class);
+    private final static String DEPLOYER_NAME = "puppet";
+
+    @Override
+    public String getDeployerName() {
+        return DEPLOYER_NAME;
+    }
+
+    @Override
+    public Deployment deploy(Deployment deployment) throws TestGridDeployerException {
+        String testPlanLocation = deployment.getDeploymentScriptsDir();
+        Utils.executeCommand("chmod -R 777 " + testPlanLocation, null);
```

Comment:
It would be better to add comments mentioning the purpose of each "executeCommand" as it will ease the pain of maintenance and readability.

Reply:
Fixed in 7e22a24